wpilibNewCommands/src/main/native/include/frc2/command/CommandScheduler.h

```diff
@@ -48,6 +49,8 @@ class CommandScheduler final : public nt::NTSendable,
   CommandScheduler& operator=(const CommandScheduler&) = delete;
 
   using Action = std::function<void(const Command&)>;
+  using InterruptAction =
+      std::function<void(const Command&, const std::optional<Command*>&)>;
```

Comment:
According to https://en.cppreference.com/w/cpp/utility/optional, "There are no optional references; a program is ill-formed if it instantiates an optional with a reference type."

Reply:
That's why there's no point in doubling the nullability. In case I wasn't clear, I was suggesting `optional<Command&>` (which has one "empty" value) rather than `optional<Command*>` (which has two).

Reply:
`optional<Command&>` is what I was talking about. I tried changing to `optional<const Command&>` locally and got the error message `error: static_assert failed due to requirement '!is_reference_v<const frc2::Command &>' "instantiation of optional with a reference type is ill-formed"`, and you can also look at https://godbolt.org/z/K5qhr8MYT.

You do have a good point about the two null values, but I'm not sure what to do about it. As was said in the original issue, passing `nullptr` to the callback would be very problematic, but since optional references can't be done and `optional<Command>` copies the `Command` object, I don't think there's a better option than never using `make_optional(nullptr)` as we've done already.

Reply:
Fair point.